src/_pytest/python_api.py

```diff
@@ -654,36 +649,20 @@ def raises(expected_exception, *args, **kwargs):
         raise TypeError(msg % type(exc))
 
     message = "DID NOT RAISE {}".format(expected_exception)
-    match_expr = None
 
     if not args:
-        if "message" in kwargs:
-            message = kwargs.pop("message")
-            warnings.warn(deprecated.RAISES_MESSAGE_PARAMETER, stacklevel=2)
-        if "match" in kwargs:
-            match_expr = kwargs.pop("match")
         if kwargs:
             msg = "Unexpected keyword arguments passed to pytest.raises: "
             msg += ", ".join(sorted(kwargs))
+            msg += "\nUse context-manager form instead?"
```

Comment:
can we instead remove the kwargs alltogether?

Reply:
We can't because we need to pass `kwargs` forward when using the function form

Reply:
in that case traceback_hide and a function that does tighten it down `

Reply:
Not sure I follow, can you elaborate please? 🤔

Reply:
just have the construction argument names match the expected keyword arguments, and let the raises context creation do the error + use tracebackhide

Reply:
You mean:

```python
if not args:    
    return RaisesContext(expected_exception, message, match=match, **kwargs)            
```

?

Reply:
something like that

Reply:
Done